src/bindings.cpp

```diff
@@ -113,27 +121,39 @@ static zval create_object(simdjson::dom::element element) /* {{{ */ {
         case simdjson::dom::element_type::NULL_VALUE :
             ZVAL_NULL(&v);
             break;
-        case simdjson::dom::element_type::ARRAY :
-            zval arr;
-            array_init(&arr);
+        case simdjson::dom::element_type::ARRAY : {
+            zend_array *arr;
+            array_init(&v);
+            arr = Z_ARR(v);
 
             for (simdjson::dom::element child : simdjson::dom::array(element)) {
                 zval value = create_object(child);
-                add_next_index_zval(&arr, &value);
+                zend_hash_next_index_insert(arr, &value);
             }
-
-            v = arr;
             break;
-        case simdjson::dom::element_type::OBJECT :
-            zval obj;
-            object_init(&obj);
+        }
+        case simdjson::dom::element_type::OBJECT : {
+            object_init(&v);
+#if PHP_VERSION_ID >= 80000
+            zend_object *obj = Z_OBJ(v);
+#endif
 
             for (simdjson::dom::key_value_pair field : simdjson::dom::object(element)) {
                 zval value = create_object(field.value);
-                add_property_zval_ex(&obj, field.key.data(), strlen(field.key.data()), &value);
```

Comment:
https://github.com/php/php-src/pull/6551#issuecomment-752196824

Fixing the use of strlen and allowing null bytes means that this would mean this now needs to check if the first byte is null, though